src/api/file/fileToContentVersion.ts

```diff
@@ -0,0 +1,49 @@
+/*
+ * Copyright (c) 2023, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import { readFile } from 'node:fs/promises';
+import { basename } from 'node:path';
+import { Connection } from '@salesforce/core';
+import { Record, SaveResult } from '@jsforce/jsforce-node';
+import FormData from 'form-data';
+
+export type ContentVersion = {
+  Title: string;
+  FileExtension: string;
+  VersionData: string;
+  /** this could be undefined outside of our narrow use case (created files) */
+  ContentDocumentId: string;
+} & Record;
+
+type ContentVersionCreateRequest = {
+  PathOnClient: string;
+  Title?: string;
+};
+
+export async function file2CV(conn: Connection, filepath: string, title?: string): Promise<ContentVersion> {
+  const cvcr: ContentVersionCreateRequest = {
+    PathOnClient: filepath,
+    Title: title,
+  };
+
+  const form = new FormData();
+  form.append('VersionData', await readFile(filepath), { filename: title ?? basename(filepath) });
+  form.append('entity_content', JSON.stringify(cvcr), { contentType: 'application/json' });
+
+  // POST the multipart form to Salesforce's API, can't use the normal "create" action because it doesn't support multipart
+  const CV = await conn.request<SaveResult>({
+    url: '/sobjects/ContentVersion',
+    headers: { ...form.getHeaders() },
+    body: form.getBuffer(),
+    method: 'POST',
+  });
+
+  const result = await conn.query<ContentVersion>(
+    `Select Id, ContentDocumentId, Title, FileExtension from ContentVersion where Id='${CV.id}'`
+  );
+  return result.records[0];
```

Comment:
Is a record guaranteed to be returned from the query?  I don't know the `/sobjects/ContentVersion` API so maybe it can be assumed.  But if it returns a non http 200 the code will continue and either fail during the query or throw a TypeError.

Reply:
it's the ID from one we just created in line 38.

http errors  would throw from conn.query (before line 48).  I can put some checks around it (or switch to singleRecordQuery, ex: CV saves but doesn't show up in the query for some reason)